File: app/src/main/python/models/organization.py
"""Organization class module"""

from models.org_base_model import BaseModel


class Organization(BaseModel):
    """ Organization model class
        Attributes:
            name (str): The name of the organization.
            head (str): The head or leader of the organization.
            phone (str): The organization's contact phone number.
            email (str): The email address associated with the organization.
            website (str): The website URL of the organization.
            customization (str): Any customization or branding information
                                related to the organization.

    """
    name = ""
    head = ""
    phone = ""
    email = ""
    website = ""
    customization = ""

    def __init__(self, *args, **kwargs):
        """ Class constructor """
        super().__init__(*args, **kwargs)

File: app/src/main/java/com/abstratsystems/org/models/Organization.kt
package com.abstratsystems.org.models

import com.google.gson.annotations.SerializedName
import okhttp3.MultipartBody
import java.time.LocalDateTime
import java.util.UUID

/**
 * Data class representing an Organization in the ORG Android app.
 *
 * @property id Unique identifier for the organization.
 * @property name of the organization.
 * @property head Name of the head of the organization
 * @property phone Organization's phone number.
 * @property email Organization's email address.
 * @property website Organization's website URL.
 * @property country Country of organization Headquarters
 * @property state State of organization Headquarters
 * @property city City of organization Headquarters
 * @property logo Customization data for the organization.
 * @property createdAt Timestamp indicating when the organization was created.
 * @property updatedAt Timestamp indicating the last update to the organization's information.
 */
data class Organization (
    @SerializedName("id") var id: String? = null,
    var name: String = "ORG",
    var head: String = "ORG",
    var phone: String = "",
    var email: String = "",
    var website: String = "",
    var country: String = "",
    var state: String = "",
    var city: String = "",
    var logo: MultipartBody.Part? = null, // This field will hold the image data,
    @SerializedName("created_at") var createdAt: String? = null,
    @SerializedName("updated_at") var updatedAt: String? = null
)